Mathlib/Data/Nat/Defs.lean

```diff
@@ -999,6 +1002,11 @@ lemma div_eq_iff_eq_of_dvd_dvd (hn : n ≠ 0) (ha : a ∣ n) (hb : b ∣ n) : n
     exact Nat.eq_mul_of_div_eq_right ha h
   · rw [h]
 
+lemma le_mul_div_add (hb : b ≠ 0) : a ≤ b * (a / b) + b - 1 := by
+  refine Nat.le_sub_of_add_le ?_
+  rw [succ_le_iff, ← Nat.mul_add_one, Nat.mul_comm, ← div_lt_iff_lt_mul (Nat.pos_iff_ne_zero.2 hb),
+    Nat.lt_add_one_iff]
+
```

Comment:
```suggestion
lemma le_mul_div_add (hb : b ≠ 0) : a ≤ b * (a / b) + b - 1 :=
  le_sub_one_of_lt <| lt_mul_div_succ a <| zero_lt_of_ne_zero hb
```

Reply:
Are you sure this lemma is needed, and you can't use `lt_mul_div_succ` or similar in your context?

Reply:
Your proof doesn't work, but you are right that I don't need this lemma